### ISSUE TITLE
ps2exe not passing options with double dash correctly

### ISSUE THREAD

**Issue by Jrprogrammer** [MScholtes/PS2EXE#54]
I have a script with an argument parser built in. 
The arg parser looks for options with a single dash `(-h)` or a double dash `(--help)`
When running the script as a ps1 file the arg parser seems to work fine but when it's compiled into an executable it seems to cut of my checks and only try for the single dashed options.

This is my code:
```powershell
switch -Regex ($args[$i]) {
    "-h\b|--help\b" {Show-Text}
    "-n\b|--name\b" {$Name = (Set-Name $args[$i+1])}
    "-ns\b|--no-shortcut\b" {$NoShortcut = $true}
    "-l\b|--list\b" {Show-Variants}
    "-i\b|--installed\b" {Show-Installed}
    "-e\b|--extensions\b" {Show-Extensions $args[$i+1]}
    default {
        if ($args[$i][0] -eq "-") {
            Write-Output "Argument not found: "$args[$i]
            exit
        } 
    }
}
```

Might it have to do something with the argument parsing from the executable itself?

**Comment by MScholtes** [comment]
Hello @Jrprogrammer,

the default way to process parameters in _Powershell_ is to pass named parameters (with a dash). Only if this does not work, position parameters are used. Since this decision can only be made at runtime of the script, I had to decide on a method of parsing for _PS2EXE_ (which is precompiling). I decided on the common method (named parameters), for the reason see the comment of @scriptingstudio).

Sorry for that, greetings

Markus
